Jenkinsfile

```diff
@@ -34,6 +34,17 @@ pipeline {
                   }
                 } // post
             } // stage ('bullshtml')
+            stage ('maldetect') {
+                steps {
+                    sh script: './get_maldetect.sh',
+                       label: 'Get maldetect'
+                }
+                post {
+                  success {
+                      archiveArtifacts artifacts: 'maldetect-current.tar.gz'
+                  }
+                } // post
+            } // stage ('maldetect')
```

Comment:
Is this basically just a poor-man's proxy?

How will this file be used in CI such that external users of our pipeline code won't be using URLs to our Jenkins server that cannot be reached outside of wolf?

Reply:
More like a cache to avoid having to go outside the local network for it for every job that is run.

The URL for these artifacts will be accessed via Jenkins environment variables, currently with the assumption that the GitHub project is named the same as on our Jenkins.

I could not find the right path syntax to get the copyArtifact plugin to work on GitHub Multibranch pipeline jobs.

Reply:
Yes, _caching_ proxy is what I meant.  I adjusted my question.

While I like the idea of using Jenkins environment variables for local/site customisation/override, they shouldn't be necessary for all sites to configure where a reasonable/typical default exists.  This seems like a good case for that.  A site that does not define the environment/global variable should just always fetch the object from it's canonical source.  Is that the intention here?

This would be the same for a local repo cache for example.  If we want to provide overrides for repos that we cache locally, that should be an optionally configurable override where no configuration means the default upstream is used.  This might not be the case as we have it currently configured, but we should correct that if so and not add to that debt.

I know there are cases where we could/should be using local repo caches more but we do not.  The Dockerfiles stand out in my mind as one place.  But again, those should work as overrides so that those Dockerfiles work outside of our environment without any configuration needing done.